private/bufpkg/bufimage/bufimagebuild/builder.go

```diff
@@ -447,9 +331,10 @@ func getImageFilesRec(
 		unusedDependencyIndexes = make([]int32, 0, len(unusedDependencyFilenames))
 	}
 	var err error
-	for i, dependency := range descFileDescriptor.GetDependencies() {
+	for i := 0; i < fileDescriptor.Imports().Len(); i++ {
```

Comment:
The old protoreflect would just return non-public, non-weak dependencies with `GetDependencies()`. Do we need to filter those out here in this loop?

Reply:
This isn't right. The old protoreflect returns all dependencies. The other accessors for public and weak return a subset of this.
https://pkg.go.dev/github.com/jhump/protoreflect@v1.13.0/desc#FileDescriptor.GetDependencies